retrive old changes

HotoRas opened the pull request:
<!-- ℹ お読みください / README
PRありがとうございます！ PRを作成する前に、コントリビューションガイドをご確認ください:
Thank you for your PR! Before creating a PR, please check the contribution guide:
https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md
-->

## What
<!-- このPRで何をしたのか？ どう変わるのか？ -->
<!-- What did you do with this PR? How will it change things? -->
- 680de20711c27843fb4127af3ddb1a03b519008b default.yml에 proxyRemoteFiles 옵션을 추가함 by @laonmofu
- nekoplanet/misskey-io#13 Add changelog by @HotoRas
- 2215de539d8bd49c9a1fd3b3cb0596c6773b1a8b by @HotoRas
- c645e8f0defd2670eaf60f51c10924caf7acb2f1 by @HotoRas
- nekoplanet/misskey-io#15 노트 수정 기능 부활 (3트) by @HotoRas
- nekoplanet/misskey-io#16 Feat: "다른 계정 추가" 버튼 아래에 "새 계정 추가" 버튼이 살아 있어서 지웠습니다 by @HotoRas
- nekoplanet/misskey-io#17 Typecheck Fix by @janghoseo
- nekoplanet/misskey-io#21 Fix note edit 2 by @HotoRas

## Why
<!-- なぜそうするのか？ どういう意図なのか？ 何が困っているのか？ -->
<!-- Why do you do it? What are your intentions? What is the problem? -->

## Additional info (optional)
<!-- テスト観点など -->
<!-- Test perspective, etc -->
closes HotoRas#1
closes HotoRas#2
closes #16

## Checklist
- [x] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [x] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [x] (If needed) Update CHANGELOG.md
- [x] (If possible) Add tests

HotoRas committed:
default.yml에 proxyRemoteFiles 옵션을 추가함 680de20711c27843fb4127af3ddb1a03b519008b

HotoRas committed:
Merge changes from old repository

nekoplanet/misskey-io#13 Add changelog by @HotoRas
2215de539d8bd49c9a1fd3b3cb0596c6773b1a8b by @HotoRas
c645e8f0defd2670eaf60f51c10924caf7acb2f1 by @HotoRas
nekoplanet/misskey-io#15 노트 수정 기능 부활 (3트) by @HotoRas
nekoplanet/misskey-io#16 Feat: "다른 계정 추가" 버튼 아래에 "새 계정 추가" 버튼이 살아 있어서 지웠습니다 by @HotoRas
nekoplanet/misskey-io#17 Typecheck Fix by @janghoseo
nekoplanet/misskey-io#21 Fix note edit 2 by @HotoRas

HotoRas commented:
각종 테스트 오류는 로그 보고 내일 고치겠읍니다.

seojangho commented:
저도 빌드해서 돌려보고 코멘트 있으면 드릴게유

HotoRas committed:
Fix major issues

- 중괄호(bracelets)가 제대로 닫히지 않아 코드 전체가 고장나는 이슈 수정
- scss 형식의 css 파트의 breaking changes 반영

HotoRas committed:
Linting (phase 1)

Manual lint application to backend, bubble, reversi and sw

HotoRas committed:
Linting (phase 2)

- Auto Lint by Lint Processor

HotoRas committed:
[skip ci] Fix cypress e2e

회원가입이 아얘 버튼이 안 뜨게 되어있는데 버튼이 뜨는 상태에서만 테스트 가능한 부분이 있었음

HotoRas committed:
Update Changelog-neko.md

IO용 패치에 대한 체인지로그 삭제

HotoRas committed:
[skip ci] Update Changelog-neko.md

HotoRas commented:
테스트 수정 완료, 린팅 완료. 리뷰 가능하고 테스트 필요합니다.
- 네코쟈라시 운영방침 변경에 따라, 본 수정사항부터는 foxtail에 푸시되지 않고 neko에 바로 푸시됩니다.
  - foxtail은 네코쟈라시의 마스터 브랜치,
  - neko는 고양이별의 마스터 브랜치로 되어 있습니다

seojangho commented:
디비 테이블 정의에서 createdAt 관련 부분은 빼거나 해야 할 것 같아요. (실제로 최초 실행 후 관리자 생성시 이 오류가 납니다.)

```
ERR  *	[api]	Internal error occurred in admin/accounts/create: column note.createdAt does not exist {
  ep: 'admin/accounts/create',
  ps: { username: 'username', password: 'password' },
  e: {
    message: 'column note.createdAt does not exist',
    code: 'QueryFailedError',
    stack: 'QueryFailedError: column note.createdAt does not exist\n' +
      '    at PostgresQueryRunner.query (/misskey/node_modules/.pnpm/typeorm@0.3.20_ioredis@5.4.1_pg@8.12.0/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
      '    at async SelectQueryBuilder.loadRawResults (/misskey/node_modules/.pnpm/typeorm@0.3.20_ioredis@5.4.1_pg@8.12.0/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2192:25)\n' +
      '    at async SelectQueryBuilder.executeEntitiesAndRawResults (/misskey/node_modules/.pnpm/typeorm@0.3.20_ioredis@5.4.1_pg@8.12.0/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2040:26)\n' +
      '    at async SelectQueryBuilder.getRawAndEntities (/misskey/node_modules/.pnpm/typeorm@0.3.20_ioredis@5.4.1_pg@8.12.0/node_modules/typeorm/query-builder/SelectQueryBuilder.js:684:29)\n' +
      '    at async SelectQueryBuilder.getMany (/misskey/node_modules/.pnpm/typeorm@0.3.20_ioredis@5.4.1_pg@8.12.0/node_modules/typeorm/query-builder/SelectQueryBuilder.js:750:25)\n' +
      '    at async UserEntityService.pack (file:///misskey/packages/backend/built/core/entities/UserEntityService.js:307:24)\n' +
      '    at async file:///misskey/packages/backend/built/server/api/endpoints/admin/accounts/create.js:71:25\n' +
      '    at async ApiCallService.call (file:///misskey/packages/backend/built/server/api/ApiCallService.js:382:20)',
    id: '7da7e65a-8551-4288-a8e3-dfa732dafd8e'
  }
}
```

아마 관련 커밋 revert하면 될것같긴 한데 저도 시간날때 볼게요.

HotoRas committed:
Create 1723585019082-noteCreatedAtMissing.js

note.createdAt이 psql 쿼리 중 테이블 상에 없어서 오류 발생, 조치를 위해 마이그레이션 추가

HotoRas committed:
Fix: basic.cy.ts

syntax error

laonmofu commented:
### 1. `column MiUserProfile.hardMutedWords does not exist`
* 해당 오류로 타임라인 및 알림, 드라이브 조회가 일체 되지 않았습니다
* 네코쟈라시는 io 포크 적용 후 다시 디마이그레이션 하는 경우이므로 일단 DB를 일부 조작하는 방향을 선택했습니다.
* 다만 네코플래닛 적용 시에는 DB 관련으로 좀 더 주의깊게 생각해볼 필요가 있겠습니다.
* 임시방편으로 DB 컬럼 일부를 조작한 뒤에는 오류가 나던 기능들은 매끄럽게 잘 동작합니다.
  
  
### 2. 오브젝트 스토리지 탭은 잘 돌아왔습니다.
* 제어판 패널이 잘 돌아왔스빈다 입력 및 저장도 잘 됩니다.
* 다만 실질적으로 일부 API가 동작하지 않습니다.
* S3로 주소 생성은 하나 실질적으로 스토리지에 등록은 안 되는 듯 합니다.
* 생성된 주소로 들어가면 클플은 404를 띄우네요.
  
  
### 3. 노트 수정 기능은 새로운 오류가 납니다.
* 노트 수정시 강제 리프래시 기능은 의도대로 잘 동작합니다.
* 다만 임시방편 정도라고 생각되어 노트 하나만 리프래시 되는 방향으로 수정되어야 할 듯 합니다.
* 더불어 노트 수정 시 기존 노트가 사라지지 않고, 별개의 노트가 새로 등록되는 듯 합니다. (새 버그)

HotoRas commented:
> ### 1. `column MiUserProfile.hardMutedWords does not exist`
> * 해당 오류로 타임라인 및 알림, 드라이브 조회가 일체 되지 않았습니다
> * 네코쟈라시는 io 포크 적용 후 다시 디마이그레이션 하는 경우이므로 일단 DB를 일부 조작하는 방향을 선택했습니다.
> * 다만 네코플래닛 적용 시에는 DB 관련으로 좀 더 주의깊게 생각해볼 필요가 있겠습니다.
> * 임시방편으로 DB 컬럼 일부를 조작한 뒤에는 오류가 나던 기능들은 매끄럽게 잘 동작합니다.

이건 io 포크가 해당 컬럼을 지워서 그렇습니다. 해당 부분은 건드린 게 없으니 냥별 적용 시 문제 없을 겁니다.

> ### 2. 오브젝트 스토리지 탭은 잘 돌아왔습니다.
> * 제어판 패널이 잘 돌아왔스빈다 입력 및 저장도 잘 됩니다.
> * 다만 실질적으로 일부 API가 동작하지 않습니다.
> * S3로 주소 생성은 하나 실질적으로 스토리지에 등록은 안 되는 듯 합니다.
> * 생성된 주소로 들어가면 클플은 404를 띄우네요.

클플이랑 주소 생성 규칙이 다른가 보네요. 제가 클플 S3를 만져보지 않았어서 잘 모르겠습니다..

> ### 3. 노트 수정 기능은 새로운 오류가 납니다.
> * 노트 수정시 강제 리프래시 기능은 의도대로 잘 동작합니다.
> * 다만 임시방편 정도라고 생각되어 노트 하나만 리프래시 되는 방향으로 수정되어야 할 듯 합니다.
> * 더불어 노트 수정 시 기존 노트가 사라지지 않고, 별개의 노트가 새로 등록되는 듯 합니다. (새 버그)

강제 리프레시는 저도 임시방편이라 생각하고 있습니다. 아얘 현재 노트로 다시 이동시키는 게 나아 보이고요.
새 노트 버그는... 백엔드를 뒤집어 까봐야 할 것 같습니다. 좀 걸리지 않을까요?

laonmofu commented:
2에 대해서는 말이 약간 헷갈렸는데, 그러니까...

1. 프론트에서 파일을 올리는 행위 자체는 문제가 없습니다. 
2. 미스키 프론트에서는 잘 올라간 취급을 합니다. 오류도 안 나고요.
3. 그런데 실질적으로 백앤드에서 해당 파일을 옵젝스토리지에 업로드하는 행동은 안 하는 것 같습니다.
4. 그래서 R2에 들어가면 해당 파일은 찾을 수 없습니다. 링크로 접근해도 404가 뜨고요.
5. 썸네일도 제공이 안 됩니다. 아예 미스키에서도 열 수가 없다고 뜨네요.

아마 근데 이도 io 포크에서 옵젝스토리지 기능이 빠졌다가 돌아온 부작용같습니다. 아마 건드린 게 없다면 네코플래닛에서는 잘 작동할테니 네코쟈라시만의 숙제가 될 것 같기도 하고요.

laonmofu commented:
노트 수정 기능의 경우 (없는 기능을 덧붙이는 것이라서) 기간이 오래 걸려도 좋으니 편하게 작업해주셨으면 좋겠습니다.

caipira113 committed:
feat: note update federation (#1)

ellelle

HotoRas commented:
노트 수정 기능에서 새 노트가 튀어나오는 문제는 `new MiNote()`를 호출하기 때문인 거로 보이는데, 확인 한 번 다시 해 볼게요. 굳이 `new MiNote()`를 안 쓰고도 가능할 것 같아서.

강제 리프레시는 페이지 전체가 새로고침되는 거니까... 노트 수정이 끝나면 작성한 노트로 이동하도록 변경할 예정입니다. `MiNote.id`값을 조회하면 되니까 큰 어려움은 없을 거에요.

HotoRas committed:
fix (frontend): 노트 업데이트 시 notes/update api를 호출하도록

기존; notes/create api만 호출하도록 되어 있었음
-> notes/update와 이를 핸들링하기 위한 부분을 추가해 랜딩

HotoRas committed:
feat (frontend): 노트 수정 후 수정된 노트로 리다이렉션되도록

HotoRas commented:
디테일한 부분은 머지 전에 수정하는 거라서 별도로 체인지로그에 적진 않을 예정입니다

다음 부분이 수정되었습니다.
- [x] 노트 업데이트 시 `notes/create` api를 호출하는 문제
  - `notes/update`로 변경 
- [x] 노트 수정 시 페이지를 새로고침하는 기능
  - 노트를 팝업으로 띄우도록 변경

yunochi committed:
[ci skip] hotfix: Undo note update

yunochi committed:
Merge pull request #4 from yunochi/munochi-dev

노트 수정기능 연합

caipira113 committed:
fix: When updating Note, emojis are not reflected through Streaming

HotoRas committed:
Fix (frontend): MkPostForm: updatedAt is not defined

and some effective changes for code

HotoRas committed:
Fix (frontend): MkPostForm: updatedAt is not defined (2)

HotoRas committed:
disable(backend-test): Disable failing tests

뭐 어떻게 하라는건데

HotoRas committed:
Update node version to 22

Local only!

HotoRas committed:
Revert to node 20

and some pnpm.lock fixes
and some package version mismatch fixes

HotoRas committed:
feat (reimplement): update to node 22

된대요

HotoRas commented:
- [x] 노트 수정 시 `[노트]` 버튼이 눌리지 않음

HotoRas committed:
Fix (frontend): MkPostForm: 노트 수정 모드 시 무조건 포스트가 가능하도록

props.editMode가 true인 경우
- posting.value not in (null | undefined | false)
- posted.value not in (null | undefined | false)
인 상태라 별도로 해결해줘야 함

HotoRas commented:
## Pull Request now Ready to Review
reference @seojangho @laonmofu

### This pull request...
- brings back old changes
  - Note Edit
    - closes HotoRas#1
    - references nekoplanet/misskey-io#5
    - references nekoplanet/misskey-io#15
    - references nekoplanet/misskey-io#21
    - references nekoplanet/misskey-io#24
  - Nekoplanet changelog
    - references nekoplanet/misskey-io#13
- Update to Node 22
  - closes HotoRas#2
  - closes #16
- Update canisue-lite (BrowsersList) and refactor `pnpm-lock` (automated)

HotoRas committed:
chore (BrowsersList): Update canisue-lite

pnpm-lock.yaml이 변경되었습니다. 빌드 전 pnpm cleanall; pnpm i 명령을 꼭 실행해주세요.

HotoRas commented:
### Lint is failing on node 22
Lint가 node 22에서 계속 실패하는 것으로 확인되어 있는데, EsLint가 node 20에서는 사용 가능하지만 node 22에서 제거된 `assert`에 여전히 의존하고 있어서 실패하는 것입니다.
이 오류는 EsLint-Typescript 모듈이 node 22에 대응하고 node 18을 지원하지 않는 업데이트를 내기 전까지 유효합니다.

HotoRas committed:
[ci skip] Update changelog

체리픽을 했으니 어디서 했는진 알려줘야 될 거 아니여

seojangho committed:
migration: set IF NOT EXISTS to column creation DDL

HotoRas committed:
bugfix (migration): set IF NOT EXISTS to column creation DDL (#4)

HotoRas committed:
Revert "migration: set IF NOT EXISTS to column creation DDL"

This reverts commit 69ed45d25bd72b9ae7690fe88a5320b09c672600.

HotoRas committed:
Revert "[ci skip] Update changelog"

This reverts commit caf9eb7c61103127e99defc1fe953fe9dd36e073.

HotoRas committed:
Revert "fix: When updating Note, emojis are not reflected through Streaming"

This reverts commit 5f7a58ad312c9afe258687bbdf75ad9c98576c18.

HotoRas committed:
Revert "feat: note update federation (#1)"

This reverts commit 6d984089eca365aefbd73c3ab2c386c6e7ae8f56.

HotoRas committed:
fix: 마이그레이션 코드가 해당 테이블이 있는지 확인하도록

안하면 터짐

HotoRas committed:
fix: double import

HotoRas committed:
fix (lint): double definition

HotoRas committed:
fix (frontend): 빼놨던 함수가 사라져서

그거 다시 가지고 옴

HotoRas commented:
# Pull request now ready to review (3rd)

staging fixes:
- Note edit function works funky
- Note edit function doesn't federate

seojangho committed:
remove duplication

seojangho committed:
revert some of c49a34811f

seojangho committed:
eliminate mixed usage of updateMode and editMode

seojangho committed:
(revert me later: temporarily)

seojangho committed:
fix updatedAtHistory is undefined

seojangho committed:
remove openDetail();

HotoRas committed:
Debug note edit (#5)

알려진 버그: canEditNote 권한이 없는 사용자에게도 "Edit" 메뉴가 보이는 상태일거에요.

그 외에는 잘 작동하는것 같음

HotoRas commented:
## Pull Request ready to Merge

### Stages
- [x] Note edit fully works
  - [x] Federate TO apub
  - [x] Federate FROM apub
  - [x] Update and go
  - [x] Hide if no perm
- [x] Node 22
  - [x] Just works
- [x] Works on their computer
- [x] Make docker image

HotoRas commented:
관련 이슈 모두 Closed as Completed로 표시했습니다. 진짜 머지만 남았네요
(도커 이미지도 있음)

HotoRas commented:
나중에 다시 (작업은 완료!)
